packages/native/src/Link.tsx

```diff
@@ -12,7 +12,10 @@ type Props<ParamList extends ReactNavigation.RootParamList> = {
   onPress?: (
     e: React.MouseEvent<HTMLAnchorElement, MouseEvent> | GestureResponderEvent
   ) => void;
-} & (TextProps & { children: React.ReactNode });
+} & (Omit<TextProps, 'disabled'> & {
+  disabled?: boolean | null;
```

Comment:
Why is this change needed? We are adding an option to pass `null`

Reply:
After the upgrade of `@types/react-native` Typescript started to whine about incompatible `disabled` props:

```
Types of property 'disabled' are incompatible.
    Type 'boolean | null | undefined' is not assignable to type 'boolean | undefined'.
      Type 'null' is not assignable to type 'boolean | undefined'.
``` 

So I've looked up on how it's figured out on `main` and simply did the same:
https://github.com/react-navigation/react-navigation/blob/main/packages/native/src/Link.tsx#L15